src/core/sorted_map.h

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <optional>
+#include <string>
+#include <vector>
+
+extern "C" {
+#include "redis/dict.h"
+#include "redis/zset.h"
+}
+
+namespace dfly {
+
+namespace detail {
+
+class SortedMap {
+ public:
+  using ScoredMember = std::pair<std::string, double>;
+  using ScoredArray = std::vector<ScoredMember>;
+
+  SortedMap();
+  ~SortedMap();
+
+  static SortedMap* FromListPack(const uint8_t* lp);
```

Comment:
The reason I return a pointer is because the context demands so. I have no use for this object on the stack because I must pass its pointer into the CompactObject.

Please note that while it looks like a new class, it's really about consolidation of dispersed code under a single entity. 
I will add an ownership comment.

Reply:
I'd return a `unique_ptr` rather than document it, it's self documenting and safer. It also won't have any performance implications (just call `.release()` from the callsite)